static/js/components/auth/AuthEmailForm.js

```diff
@@ -2,49 +2,114 @@
 /* global SETTINGS:false */
 import React from "react"
 import ReCAPTCHA from "react-google-recaptcha"
+import styled from "styled-components"
+import debounce from "lodash/debounce"
 
 import { validationMessage } from "../../lib/validation"
 
 import type { EmailForm } from "../../flow/authTypes"
 import type { FormProps } from "../../flow/formTypes"
 
-const AuthEmailForm = ({
-  form,
-  validation,
-  onSubmit,
-  onUpdate,
-  onRecaptcha,
-  processing
-}: FormProps<EmailForm>) => {
-  return (
-    <form onSubmit={onSubmit} className="form">
-      <div className="emailfield row">
-        <input
-          type="email"
-          name="email"
-          value={form.email}
-          onChange={onUpdate}
-          placeholder="Email"
-        />
-        {validationMessage(validation.email)}
-      </div>
-      {SETTINGS.recaptchaKey && onRecaptcha ? (
-        <div className="recaptcha row">
-          <ReCAPTCHA sitekey={SETTINGS.recaptchaKey} onChange={onRecaptcha} />
-          {validationMessage(validation.recaptcha)}
+type Props = FormProps<EmailForm>
+type State = {
+  recaptchaScale: number
+}
+
+// These needs to stay in sync with ReCAPTCHA's runtime width
+const RECAPTCHA_WIDTH = 304
+const RECAPTCHA_HEIGHT = 78
+
+const StyledReCAPTCHA = styled(ReCAPTCHA)`
+  transform: scale(${props => props.scale.toFixed(3)});
+  transform-origin: 0 0;
+  height: ${props => (props.scale * RECAPTCHA_HEIGHT).toFixed(0)}px;
+`
+
+class AuthEmailForm extends React.Component<Props, State> {
+  recaptcha: null
+
+  constructor(props: Props) {
+    super(props)
+    this.recaptcha = null
+    // only rescale up to 4x a second to salvage some performance
+    this.scaleRecaptcha = debounce(this.scaleRecaptcha.bind(this), 250)
+    this.state = {
+      recaptchaScale: 1.0
+    }
+  }
+
+  componentDidMount() {
+    window.addEventListener("resize", this.scaleRecaptcha)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener("resize", this.scaleRecaptcha)
+  }
+
+  scaleRecaptcha = () => {
```

Comment:
since we're binding this in the constructor, should it just be a standard class method instead of an arrow-function class property?

Reply:
@aliceriot I'm going to back that out, since I'm getting this `flow` error now:

```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ static/js/components/auth/AuthEmailForm.js:35:10
Cannot assign debounce(...) to this.scaleRecaptcha because property
scaleRecaptcha is not writable.
     32│     super(props)
     33│     this.recaptcha = null
     34│     // only rescale up to 4x a second to salvage some performance
     35│     this.scaleRecaptcha = debounce(this.scaleRecaptcha.bind(this), 250)
     36│     this.state = {
     37│       recaptchaScale: 1.0
     38│     }
```

Reply:
yeah just saw that on the CI build

Reply:
seems like that's what we want to do!